File: go/difference-of-squares/difference_of_squares.go
package diffsquares

//SquareOfSum returns the square of the sum of the first ten natural numbers
func SquareOfSum(n int) int {
	sum := (1 + n) * n / 2
	return sum * sum
}

//SumOfSquares return the sum of the squares of the first ten natural numbers
func SumOfSquares(n int) int {
	return n * (n + 1) * (2*n + 1) / 6
}

//Difference returns the difference of SquareOfSum and SumOfSquares
func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}

File: ruby/difference-of-squares/difference_of_squares.rb
# frozen_string_literal: true

class Squares
  def initialize(n)
    @n = n
  end

  def square_of_sum
    sum = (1 + @n) * @n / 2
    sum * sum
  end

  def sum_of_squares
    @n * (@n + 1) * (2 * @n + 1) / 6
  end

  def difference
    square_of_sum - sum_of_squares
  end
end